plugin.cpp

```diff
@@ -500,6 +505,84 @@ void plugin_ingest(PLUGIN_HANDLE *handle,
 			}
 
 		}
+		else if (assetAction->actn == action::SPLIT)
```

Comment:
Doing this on every ingest call is expensive and not necessary. Pre-parse the document and save the data in a more efficient form for the use of ingest. You only need to do the expensive JSON parsing and traversal of the document when the configuration is changed, which is much less frequently than when data is ingested.

Reply:
@MarkRiddoch 
Entire logic is written in single file that is plugin.cpp. It will be easier to maintain and make code more flexible for updates in future as well. 
Is it ok if I refactor it and introduce a class to perform all the required operations like remove/include/split etc? Please suggest.